Lab_1/Aula_rest/src/macros.h

```diff
@@ -21,8 +21,8 @@ enum stateMachine {Start, FLAG_RCV, A_RCV, C_RCV, BCC_OK, DONE};
 #define C_SET 0b00000011 // (0x03) Campo de Controlo - SET (set up)
 #define C_DISC 0b00001011 // (0x0B) Campo de Controlo - DISC (disconnect)
 #define C_UA 0b00000111 // (0x07) Campo de Controlo - UA (Unnumbered Acknowledgement)
-#define C_RR(r) ((0b10000101) &= (r) << (7)) // (0x05 OU 0x85) Campo de Controlo - RR (receiver ready / positive ACK))
-#define C_REJ(r) ((0b10000001) &= (r) << (7)) // (0x01 OU 0x81) Campo de Controlo - REJ (reject / negative ACK))
+#define C_RR(r) ((0b10000101) & (r) << (7)) // (0x05 OU 0x85) Campo de Controlo - RR (receiver ready / positive ACK))
+#define C_REJ(r) ((0b10000001) & (r) << (7)) // (0x01 OU 0x81) Campo de Controlo - REJ (reject / negative ACK))
 #define C_I(r) ((0b01000000) & (r) << (6)) // (0x00 0x40) Campo de Controlo - Tramas I 
 
```

Comment:
`(r) << (7)` precisa de ser `((r) << (7))` ou << tem precedência a &?

Reply:
https://en.cppreference.com/w/c/language/operator_precedence
Shift feito primeiro que AND, no problem here
![image](https://user-images.githubusercontent.com/54408098/96337505-8ef14680-107f-11eb-88be-9778051e1609.png)